bolt-taproot-gossip.md

```diff
@@ -0,0 +1,1083 @@
+   along with a [BIP86 tweak][bip86-tweak]. This provides a slightly weaker
+   binding to the LN context than legacy channels do but at least somewhat 
+   limits how the output can be spent due to the script-path being disabled.
+   The context binding with legacy channels is greater because an attacker 
+   trying to create gossip spam would need to create 2-of-2 multisig P2WSH 
+   outputs on-chain that would require them to then pay for the bytes of two
+   signatures at the time of spending the output. This extra cost provided some
+   extra deterrence for attackers. This deterrence is not present with P2TR 
+   transactions.
+3. The owners of `bitcoin_key_1` and `bitcoin_key_2` agree to be associated with  
+   a channel owned by `node_1` and `node_2`.
+4. The owners of `node_ID_1` and `node_ID_2` agree to being associated with the 
+   output paying to `bitcoin_key_1` and `bitcoin_key_2`. 
+
+For legacy channels, these last two proofs are made possible by including four
+separate signatures in the `channel_announcement`. However, [BIP-340][bip-340]
+signatures and the MuSig2 protocol allow us to now aggregate these four
+signatures into a single one. Verifiers will then be able to aggregate the four
+keys (`bitcoin_key_1`, `bitcoin_key_2`, `node_ID_1` and `node_ID_2`) using
+MuSig2 key aggregation and then they can do a single signature verification
+check instead of four individual checks.
+
+## Timestamp fields
+
+In this document, the `timestamp` fields in messages are block heights instead
+of the UNIX timestamps used in the legacy gossip messages. 
+
+### Rate Limiting
+
+A block height based timestamp results in more natural rate limiting for gossip
+messages: nodes are allowed to send at most one announcement and update per
+block. To allow for bursts, nodes are encouraged not to use the latest block
+height for their latest announcements/updates but rather to backdate and use
+older block heights that they have not used in an announcement/update. There of
+course needs to be a limit on the start block height that the node can use:
+for `channel_update_2` messages, the first timestamp must be the block height in
+which the channel funding transaction was mined and all updates after the
+initial one must have increasing timestamps. Nodes are then responsible for
+building up their own timestamp buffer: if they want to be able to send
+multiple `channel_update_2` messages per block, then they will need to ensure 
+that there are blocks during which they do not broadcast any updates. This 
+provides an incentive for nodes not to spam the network with too many updates.
+
+To also prevent nodes from building up too large of a burst-buffer with which 
+they can spam the network and to give a limit to how low the block height on a 
+`node_announcement_2` can be: nodes should not be allowed to use a block height
+smaller 1440 (~ a day worth of blocks) below the current block height.
```

Comment:
```suggestion
smaller 144 (~ a day worth of blocks) below the current block height.
```

Reply:
I guess the sweet-spot value has to be something low enough to battle spam, but also sufficient for legit gossip messages? Could this depend on the size of the node?

I would expect a big routing node with 100s of channels to produce more gossip than a single-channel node.